runtime/context.go

```diff
@@ -23,7 +28,20 @@ func AnnotateContext(ctx context.Context, req *http.Request) context.Context {
 			pairs = append(pairs, key[len(metadataHeaderPrefix):], val[0])
 		}
 		if key == "Authorization" {
-			pairs = append(pairs, key, val[0])
+			pairs = append(pairs, "authorization", val[0])
+		}
+	}
+	if req.Header.Get(xForwardedHost) != "" {
+		pairs = append(pairs, strings.ToLower(xForwardedHost), req.Header.Get(xForwardedHost))
+	} else if req.Host != "" {
```

Comment:
I guess you need to append `req.Host` even if you already have `X-Forwarded-For` header.